wp-content/themes/citylimits/functions.php

```diff
@@ -332,38 +332,55 @@ function citylimits_google_analytics() {
 
 			ga( 'create', 'UA-529003-1', 'auto' );
 			ga( 'send', 'pageview' )
-		</script>
-
-		<?php
-		global $wp_query;
-		if ( is_single() ) {
-			// Get all registered taxonomies
-			$taxonomies = get_taxonomies();
-
-			$excluded_taxonomies = array(
-				'nav_menu',
-				'link_category',
-			);
-
-			foreach ( $taxonomies as $taxonomy ) {
-				// Skip this taxonomy if it's on the excluded list
-				if ( in_array( $taxonomy, $excluded_taxonomies ) ) {
-					continue;
-				}
-			}
 
-			$categories = get_the_category();
-			if ( $categories ) {
-				foreach ( $categories as $category ) {
-					if ( ! empty ( $category->cat_name ) ){
-						echo “ga( ‘set’, contentGroup2, ‘“ . $category->cat_name . “’ );\n”;
+			<?php
+			global $wp_query;
+			if ( is_single() ) {
+
+				$taxonomies = array(
+					1 => 'category',
+					2 => 'neighborhoods',
+					3 => 'prominence',
+					4 => 'post-type',
+					5 => 'series',
+				);
+
+				foreach ( $taxonomies as $tax_key => $taxonomy ) {
+
+					$terms = get_terms( array(
+					    'taxonomy' => $taxonomy,
+					    'hide_empty' => false,
+					) );
+					foreach ( $terms as $term ) {
+						if ( ! empty ( $term->name ) ){
+							echo "ga( 'set', contentGroup" . $tax_key . ", '" . esc_attr( $term->name ) . "' );\n";
```

Comment:
For categories, this sets contentGroup1 to _every category on the site_.

Here's a sample from post 1194193, http://citylimits.org/2017/01/31/review-process-begins-in-the-rockaways-for-de-blasios-2nd-neighborhood-rezoning/ using the staging branch on vagrant:

```

			( function ( i, s, o, g, r, a, m ) {i['GoogleAnalyticsObject']=r;i[r]=i[r]|| function() {( i[r].q=i[r].q||[] ).push( arguments )},i[r].l=1*new Date();a=s.createElement( o ), m=s.getElementsByTagName( o )[0];a.async=1;a.src=g;m.parentNode.insertBefore( a, m )} )
			( window,document,'script','https://www.google-analytics.com/analytics.js','ga' );

			ga( 'create', 'UA-529003-1', 'auto' );
			ga( 'send', 'pageview' )

			ga( 'set', contentGroup1, 'ARTS and CULTURE' );
ga( 'set', contentGroup1, 'BRONX BUREAU' );
ga( 'set', contentGroup1, 'Bronx News Network' );
ga( 'set', contentGroup1, 'BROOKLYN BUREAU' );
ga( 'set', contentGroup1, 'BUREAUS' );
ga( 'set', contentGroup1, 'CITY VIEWS: OPINIONS and ANALYSIS' );
ga( 'set', contentGroup1, 'CITY WIRE: THE BLOG' );
ga( 'set', contentGroup1, 'Citywide' );
ga( 'set', contentGroup1, 'CommunityWire' );
ga( 'set', contentGroup1, 'ECONOMY' );
ga( 'set', contentGroup1, 'EDUCATION' );
ga( 'set', contentGroup1, 'GOVERNMENT' );
ga( 'set', contentGroup1, 'HEALTH and ENVIRONMENT' );
ga( 'set', contentGroup1, 'HOUSING and DEVELOPMENT' );
ga( 'set', contentGroup1, 'Investigations' );
ga( 'set', contentGroup1, 'JUSTICE' );
ga( 'set', contentGroup1, 'Manhattan' );
ga( 'set', contentGroup1, 'Max &amp; Murphy Podcasts' );
ga( 'set', contentGroup1, 'NEWS' );
ga( 'set', contentGroup1, 'NYC Snapshot' );
ga( 'set', contentGroup1, 'Opinion' );
ga( 'set', contentGroup1, 'Queens' );
ga( 'set', contentGroup1, 'Staten Island' );
ga( 'set', contentGroup1, 'The Work Site' );
ga( 'set', contentGroup1, 'Uncategorized' );
ga( 'set', contentGroup1, 'UrbaNERD' );
ga( 'set', contentGroup1, 'VIDEO' );
ga( 'set', contentGroup2, 'Bay Street' );
ga( 'set', contentGroup2, 'Bushwick' );
ga( 'set', contentGroup2, 'Chinatown' );
ga( 'set', contentGroup2, 'Downtown Far Rockaway' );
ga( 'set', contentGroup2, 'East Harlem' );
ga( 'set', contentGroup2, 'East New York' );
ga( 'set', contentGroup2, 'Flushing West' );
ga( 'set', contentGroup2, 'Gowanus' );
ga( 'set', contentGroup2, 'Inwood' );
ga( 'set', contentGroup2, 'Jerome Avenue' );
ga( 'set', contentGroup2, 'Long Island City' );
ga( 'set', contentGroup2, 'Southern Boulevard' );
ga( 'set', contentGroup3, 'Featured in Category' );
ga( 'set', contentGroup3, 'Featured in Series' );
ga( 'set', contentGroup3, 'Footer Featured Widget' );
ga( 'set', contentGroup3, 'Homepage Featured' );
ga( 'set', contentGroup3, 'Homepage Top Story' );
ga( 'set', contentGroup3, 'Homepage Top Story' );
ga( 'set', contentGroup3, 'Sidebar Featured Widget' );
ga( 'set', contentGroup4, 'Documents' );
ga( 'set', contentGroup4, 'Events' );
ga( 'set', contentGroup4, 'News' );
ga( 'set', contentGroup4, 'Opinion' );
ga( 'set', contentGroup4, 'Photos' );
ga( 'set', contentGroup4, 'Videos' );
ga( 'set', contentGroup4, 'ZoneIn En Español' );
ga( 'set', contentGroup5, '1095 Days: New Orleans after Katrina (2008)' );
ga( 'set', contentGroup5, '11237: One Neighborhood in the Bloomberg Era (2009)' );
ga( 'set', contentGroup5, 'A Risky Play: Artificial Turf in New York (2010)' );
ga( 'set', contentGroup5, 'A Tough Age: Youth in Post-Recession NY (2010)' );
ga( 'set', contentGroup5, 'Aging in New York' );
ga( 'set', contentGroup5, 'Arson Today' );
ga( 'set', contentGroup5, 'Art at the Limits' );
ga( 'set', contentGroup5, 'Atlantic Yards (2012)' );
ga( 'set', contentGroup5, 'Barriers to Getting Buff: Access to Exercise in the Bronx' );
ga( 'set', contentGroup5, 'Behind Bars: Sexual Abuse of Female Inmates (2011)' );
ga( 'set', contentGroup5, 'Beyond Promises: Bloomberg and Homelessness (2013)' );
ga( 'set', contentGroup5, 'Bringing it Home: The Affordability Battles of 2015' );
ga( 'set', contentGroup5, 'Broken Adoptions (2013)' );
ga( 'set', contentGroup5, 'Bronx Investigative Internship Program' );
ga( 'set', contentGroup5, 'Brooklyn Borough Presidency (2013)' );
ga( 'set', contentGroup5, 'Brooklyn: The Borough Behind The Brand (2011)' );
ga( 'set', contentGroup5, 'Building Justice' );
ga( 'set', contentGroup5, 'Buy or Bust: NYC&#039;s War on Drugs (2009)' );
ga( 'set', contentGroup5, 'Campaign Consultants (2013)' );
ga( 'set', contentGroup5, 'Certificates of Confusion' );
ga( 'set', contentGroup5, 'Charter Revision' );
ga( 'set', contentGroup5, 'CityPlate' );
ga( 'set', contentGroup5, 'Closing Rikers' );
ga( 'set', contentGroup5, 'Community Boards in the Spotlight' );
ga( 'set', contentGroup5, 'CommunityWire' );
ga( 'set', contentGroup5, 'Courting Crisis: The Family Court System (2012)' );
ga( 'set', contentGroup5, 'Death&#039;s Disparities: Health Inequality in New York City' );
ga( 'set', contentGroup5, 'Diversity Delayed: The CIty&#039;s MWBE Program (2013)' );
ga( 'set', contentGroup5, 'Exit Strategy: The Graduation Challenge (2008)' );
ga( 'set', contentGroup5, 'Firefighter Fatalities (2011)' );
ga( 'set', contentGroup5, 'Five-Borough Ballot (2013)' );
ga( 'set', contentGroup5, 'Forty Years of City Limits' );
ga( 'set', contentGroup5, 'Freedom and Fear: Civil Liberties in NYC (2008)' );
ga( 'set', contentGroup5, 'Getting Change: Campaign Finance Reform, 20 Years Later (2008)' );
ga( 'set', contentGroup5, 'Green Jobs Gone Missing' );
ga( 'set', contentGroup5, 'Gun Politics (2012)' );
ga( 'set', contentGroup5, 'Hard Costs: The Rising Price of Affordable Housing (2007)' );
ga( 'set', contentGroup5, 'Harlem Children&#039;s Zone (2010)' );
ga( 'set', contentGroup5, 'Homecoming with Hurdles: The hunt for housing, work and healthcare after incarceration' );
ga( 'set', contentGroup5, 'Justice Gets Evicted: Bronx Housing Court Meets a City&#039;s Crisis' );
ga( 'set', contentGroup5, 'Limits of Protection: A Child Welfare Reality Check' );
ga( 'set', contentGroup5, 'Lunchtime Tuesday: Hunger in Brooklyn' );
ga( 'set', contentGroup5, 'Making NYCHA: The people behind the policies' );
ga( 'set', contentGroup5, 'Mom and Popeyes: Small Businesses Fight to Survive (2010)' );
ga( 'set', contentGroup5, 'Monitoring the Monitors' );
ga( 'set', contentGroup5, 'New York City&#039;s Silent Sewage Crisis (2007)' );
ga( 'set', contentGroup5, 'New York&#039;s Trash Challenge' );
ga( 'set', contentGroup5, 'No Backspace' );
ga( 'set', contentGroup5, 'NYC in Congress (2012)' );
ga( 'set', contentGroup5, 'Occupy Wall Street' );
ga( 'set', contentGroup5, 'On the Border of Justice' );
ga( 'set', contentGroup5, 'Participatory Budgeting (2012)' );
ga( 'set', contentGroup5, 'Payday Loans (2012)' );
ga( 'set', contentGroup5, 'Placing Our Bets: Gambling in NY (2013)' );
ga( 'set', contentGroup5, 'Prisoners Dilemma: The Bail System (2007)' );
ga( 'set', contentGroup5, 'Private Contractors (2011)' );
ga( 'set', contentGroup5, 'Public Housing&#039;s Last Stand (2009)' );
ga( 'set', contentGroup5, 'Reading Room: Libraries at a Crossroads' );
ga( 'set', contentGroup5, 'Remember Poverty? (2011)' );
ga( 'set', contentGroup5, 'Searching for El Primero: Latinos &amp; the Mayoralty (2012)' );
ga( 'set', contentGroup5, 'Security Guards' );
ga( 'set', contentGroup5, 'Stopping the Slide' );
ga( 'set', contentGroup5, 'Straphangers Speak' );
ga( 'set', contentGroup5, 'Superstorm Sandy' );
ga( 'set', contentGroup5, 'Target of Opportunities (2008)' );
ga( 'set', contentGroup5, 'The 2009 Election' );
ga( 'set', contentGroup5, 'The Black Depression (2010)' );
ga( 'set', contentGroup5, 'The Bronx Votes' );
ga( 'set', contentGroup5, 'The Bronx Votes: 2016' );
ga( 'set', contentGroup5, 'The Chamber: A Week in the Life of the New York City Council (2009)' );
ga( 'set', contentGroup5, 'The Class of 2013' );
ga( 'set', contentGroup5, 'The Cost of Our Water' );
ga( 'set', contentGroup5, 'The Epicenter of Stop and Frisk (2012)' );
ga( 'set', contentGroup5, 'The Lessons of Willets Point (2013)' );
ga( 'set', contentGroup5, 'The Phantom Landlord (2012)' );
ga( 'set', contentGroup5, 'The Rap Sheet Trap' );
ga( 'set', contentGroup5, 'The UnPlanned City (2011)' );
ga( 'set', contentGroup5, 'The Urban Agenda (2012)' );
ga( 'set', contentGroup5, 'The Work Site' );
ga( 'set', contentGroup5, 'Three-Quarter Homes (2012)' );
ga( 'set', contentGroup5, 'Trucks (2012)' );
ga( 'set', contentGroup5, 'Unequal NY (2013)' );
ga( 'set', contentGroup5, 'Waste to Energy (2012)' );
ga( 'set', contentGroup5, 'ZoneIn' );
```